lib/tree.rb

```diff
@@ -18,40 +18,139 @@ def initialize
 
   # Time Complexity: 
   # Space Complexity: 
-  def add(key, value)
-    raise NotImplementedError
+  def add(key, value = 0)
+    # there may or may not be a root
+    # if there's no root then the key/value will be the new root
+    # otherwise check to see if value is bigger or smaller than the root value
+    # initialize new node with the key and value
+    # check if it's smaller or larger and if it's the last node of the branch
+    # if it's the last node then add it after that node
+    input = TreeNode.new(key,value)
+    if @root == nil 
+      @root = input
+      return
+    else here = @root
+    end
+    while here != nil
+      if input.key < here.key
+        if here.left != nil
+          here = here.left
+        else here.left = input
```

Comment:
This works, just it's not standard coding style.

```suggestion
        else 
          here.left = input
```